lib/web_ui/lib/src/ui/painting.dart

```diff
@@ -508,19 +570,29 @@ void decodeImageFromPixels(
   int? targetHeight,
   bool allowUpscaling = true,
 }) {
-  final Future<Codec> codecFuture = _futurize((engine.Callback<Codec> callback) {
-    return _instantiateImageCodec(
+  if (engine.experimentalUseSkia) {
+    engine.skiaInstantiateImageCodec(
       pixels,
-      callback,
-      width: width,
-      height: height,
-      format: format,
-      rowBytes: rowBytes,
+      (Codec codec) {
+        codec.getNextFrame().then((FrameInfo info) {
+          callback(info.image);
+        });
+      },
+      width,
+      height,
+      format.index,
+      rowBytes,
     );
-  });
-  codecFuture
-      .then((Codec codec) => codec.getNextFrame())
-      .then((FrameInfo frameInfo) => callback(frameInfo.image));
+    return;
+  }
+
+  Null Function(Codec) callbacker = (Codec codec) {
+    codec.getNextFrame().then((FrameInfo frameInfo) {
+      callback(frameInfo.image);
+    });
```

Comment:
The signature says it returns `Null` but the body of the function looks like it's `void`.

Reply:
Done